middle_end/flambda2/term_basics/alloc_mode.ml

```diff
@@ -12,41 +12,46 @@
 (*                                                                        *)
 (**************************************************************************)
 
-type t =
-  | Heap
-  | Local
+module For_types = struct
+  type t =
+    | Heap
+    | Local
+    | Heap_or_local
 
-type without_region = t
+  let print ppf t =
+    match t with
+    | Heap -> Format.pp_print_string ppf "Heap"
+    | Local -> Format.pp_print_string ppf "Local"
+    | Heap_or_local -> Format.pp_print_string ppf "Heap_or_local"
 
-let print ppf t =
-  match t with
-  | Heap -> Format.pp_print_string ppf "Heap"
-  | Local -> Format.pp_print_string ppf "Local"
+  let compare t1 t2 =
+    match t1, t2 with
+    | Heap, Heap | Local, Local | Heap_or_local, Heap_or_local -> 0
+    | Heap, (Local | Heap_or_local) -> -1
+    | (Local | Heap_or_local), Heap -> 1
+    | Local, Heap_or_local -> -1
+    | Heap_or_local, Local -> 1
 
-let compare t1 t2 =
-  match t1, t2 with
-  | Heap, Heap | Local, Local -> 0
-  | Heap, Local -> -1
-  | Local, Heap -> 1
+  let heap = Heap
 
-let heap = Heap
+  let local = Local
 
-let local () =
-  if Flambda_features.stack_allocation_enabled () then Local else Heap
+  let unknown = Heap_or_local
```

Comment:
I was expecting `local` and `unknown` to have the check about `Flambda_features.stack_allocation_enabled`.

Reply:
Fixed